### PR TITLE
Update Learner Classroom Assignments every 30 seconds

### LINKED ISSUE #5047: learn pages (Classroom Assignment Page) should auto-refresh to see newly activated/assigned content
### Observed behavior

learner pages doesn't auto-refresh

### Expected behavior

when a coach activates a lesson or quiz, or assigns it to  a learner, or changes its name, they should see the changes without needing to refresh

### User-facing consequences

learners might have trouble seeing content assigned during a class

### Steps to reproduce

* as a learner, open a class page
* as a coach, assign and activate a lesson and a quiz

### Context

0.12.0 beta 4

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Sets up a polling function to check the `learnerclassroom` endpoint and update Vuex if there are changes.
1. A string fix
1. Updating to v-slot syntax in a component

### Reviewer guidance

Try all of the below
1. Activate a new lesson/quiz -> lesson/quiz should appear
1. Deactivate any lesson -> lesson should disappear
1. Deactivate an un-submitted quiz -> quiz should disappear
1. Deactivate a submitted quiz -> quiz should not disappear
1. Rename a lesson/quiz -> new name should appear
1. Change the resources in a lesson -> progress bar should change accordingly

### References

Fixes #5047 
----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/views/classes/ClassAssignmentsPage.vue [hunk 1]
```diff
@@ -52,6 +53,16 @@
         lessons: state => state.currentClassroom.assignments.lessons,
       }),
     },
+    mounted() {
+      this.pollForUpdates();
+    },
+    methods: {
+      pollForUpdates() {
+        this.$store.dispatch('classAssignments/updateWithChanges').then(() => {
+          setTimeout(this.pollForUpdates, 30000);
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/views/classes/ClassAssignmentsPage.vue]: 1. should these be reversed so the timeout happens first and the poll doesn't occur immediately on `mounted`?
2. we should probably assign the ID returned from `setTimeout` to a variable in `data`. Then on `beforeDestroy` we can call e.g. `clearTimeout(timeoutId)`

**Review by indirectlylit — COMMENTED** [review]
looks good.

1. let's make sure to clear the timeout on destroy
2. looks like a merge conflict

File: kolibri/plugins/learn/assets/src/views/classes/ClassAssignmentsPage.vue (hunk 1 above)
Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/views/classes/ClassAssignmentsPage.vue]: Could we file a follow up issue from this to look at optimizing the backend piece of this to reduce server load?

Comment (indirectlylit) [reply kolibri/plugins/learn/assets/src/views/classes/ClassAssignmentsPage.vue]: https://github.com/learningequality/kolibri/issues/5757

**Review by indirectlylit — APPROVED** [review]

**Pull request merged**